codalab/worker/worker.py

```diff
@@ -476,9 +476,13 @@ def process_runs(self):
         for uuid in self.runs:
             prev_state = self.runs[uuid]
             self.runs[uuid] = self.run_state_manager.transition(prev_state)
+            # it's possible that a bundle can return to the same stage, so we don't want to reset that initial stage.
```

Comment:
I think this would be a clearer comment that could replace the two comments you have here (it's currently not clear what "reset" means until you actually read & understand the code):
"Only start keeping stats for a new stage when the run has actually transitioned to that stage."